src/main/java/frc/robot/subsystems/scoring/ScoringSubsystem.java

```diff
@@ -0,0 +1,128 @@
+package frc.robot.subsystems.scoring;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class ScoringSubsystem extends SubsystemBase {
+    private final ShooterIO shooterIo;
+    private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+    private final AimerIO aimerIo;
+    private final AimerIOInputsAutoLogged aimerInputs = new AimerIOInputsAutoLogged();
+
+    private final Timer timer = new Timer();
+
+    private enum ScoringState {
+        WAIT,
+        INTAKE,
+        PRIME,
+        SHOOT,
+        ENDGAME
+    }
+
+    public enum ScoringAction {
+        INTAKE,
+        AIM,
+        SHOOT,
+        ABORT,
+        ENDGAME
+    }
+
+    private ScoringState state = ScoringState.WAIT;
+
+    public ScoringSubsystem(ShooterIO shooterIo, AimerIO aimerIo) {
+        this.shooterIo = shooterIo;
+        this.aimerIo = aimerIo;
+    }
+
+    public void setAction(ScoringAction action) {
+        switch (action) {
+            case INTAKE:
+                if (state == ScoringState.WAIT) {
+                    state = ScoringState.INTAKE;
+
+                    aimerIo.setAimAngRad(0);
+                    shooterIo.setShooterVelRPM(-10);
+                    shooterIo.setKickerVolts(-1);
+                }
+                break;
+            case AIM:
+                if (state == ScoringState.WAIT && true) { // TODO: Add banner sensor
+                    state = ScoringState.PRIME;
+
+                    shooterIo.setShooterVelRPM(100);
+                }
+                break;
+            case SHOOT:
+                if (state == ScoringState.PRIME) {
+                    state = ScoringState.SHOOT;
+
+                    shooterIo.setKickerVolts(1);
+                    timer.reset();
+                    timer.start();
+                }
+                break;
+            case ABORT:
+                state = ScoringState.WAIT;
+
+                aimerIo.setAimAngRad(0);
+                shooterIo.setShooterVelRPM(0);
+                shooterIo.setKickerVolts(0);
+                break;
+            case ENDGAME:
+                if (state == ScoringState.WAIT) {
+                    state = ScoringState.ENDGAME;
+                }
+                break;
+        }
+    }
+
+    private void abort() {}
```

Comment:
For what it's worth, this function will need to set the shooter velocity to 0 otherwise it will keep running

Reply:
Setting it here will keep setting 0 constantly though, I added it to the action to only run once

Reply:
I don't think it's a problem if it runs every cycle fwiw. If it's a computational issue (which would be bad for when we actually want to run the shooter), we can add a new state that turns everything off and then instantly transitions to a no-op state.

If we refactor the actions as suggested, the current setup would be an anti-pattern because actions will just be a setter. I'm suggesting implementing Actions the current way to make them an added boolean for determining the next state